geospyer/geospy.py

```
@@ -90,86 +290,17 @@ def locate_with_gemini(self,
                 "exception": str        # Optional exception information
             }
         """
+        logger.info(f"Starting geolocation analysis for: {image_path}")
+        
         # Convert image to base64
         try:
-            image_base64 = self.encode_image_to_base64(image_path)
+            image_base64, mime_type = self.encode_image_to_base64(image_path)
         except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)